holder.go

```diff
@@ -38,6 +38,9 @@ const (
 
 	// fileLimit is the maximum open file limit (ulimit -n) to automatically set.
 	fileLimit = 262144 // (512^2)
+
+	// notNullFieldName is the name of the internal field used to store not-null values.
+	notNullFieldName = "notnull" // TODO: use a name less likely to collide?
```

Comment:
errrr - is it possible to give this a name that wouldn't even be accepted as a field name by our API? 

I'm imagining some obscure security problem where a bad actor is able to craft some data so as to collide with the notnull field name and cause a crash or incorrect results.

Reply:
I'll test a few options. I don't like the garbage character idea. I would prefer something like `_notnull`, but we had discussed making it queryable as a normal field, so that may work internally, but not be queryable from the outside.

Reply:
oh, good point about querying it directly - `_notnull` is pretty good because it wouldn't be createable as a field via the API due to the leading underscore, but is still easily readable. It looks like we only do the validation on field creation, so queries wouldn't be affected.

What do you think about `_exists`?

Reply:
we will need some way to bypass the validation in `NewField` in field.go though.

Reply:
Regarding the "bad actor" concern, there's currently nothing in place to prevent an end user from writing directly to the `notnull` field. I didn't think it was critical to lock that down by preventing external writes, but I can do that if you think it's necessary.

Reply:
unless we need to completely lock it down, I could make it something reasonably odd like `notnull__`

Reply:
oops, I posted those last two comments before I saw your earlier comments. I'm ok with `_exists` but I'll need to bypass the validate (as you said). Let me try that.

Reply:
That name implies that you don't like "notnull". Are you suggesting that the config option be changed to something like `TrackExistence`? (sounds a little sci-fi)

Reply:
ha, I've just taken another look through the code, and I do think I like exists/existence in place of notnull. It's not a super strong opinion, but I do think it reads a bit better in most cases.

Reply:
I took a stab at bypassing field name validation in order to support something like `_notnull`, but because of the layers of field-creation functions that have to be considered, it feels a little clunky. Also, in its current state, the parser raises an error on a query like `Row(_notnull=0)`, so that would need to be addressed as well. Overall, this is starting to feel pretty invasive, and it may be worth considering a different approach.

You can see the trial commit here: https://github.com/travisturner/pilosa/commit/410c800c1d4a938a4d03b6ae906263fd78c147eb